app/redux/wallet/reducer.js

```diff
@@ -86,6 +71,22 @@ const reducer = (state: StoreStateType = initialState, action: Action) => {
       const { transactions } = action.payload;
       return { ...state, transactions };
     }
+    case SET_LAST_USED_ADDRESSES: {
+      const { transactions } = action.payload;
+      const getFirstUniqueTransactions = (txList: TxList, n: number = 3): Contact[] => {
+        const unique = new Set();
+        for (let i = 0; i < txList.length; i += 1) {
+          if (!unique.has(txList[i])) {
+            unique.add(txList[i]);
+          }
+          if (unique.size === n) {
```

Comment:
this is ugly, add i < n to stop loop condition

Reply:
np